tensorflow/java/src/main/java/org/tensorflow/Operation.java

```diff
@@ -97,6 +99,46 @@ public Output output(int idx) {
     return new Output(this, idx);
   }
 
+  @Override
+  public int hashCode() {
+    Graph.Reference r = graph.ref();
+    try {
+      // xor the first and last 32 bits of the underlying 64-bit pointer value.
+      return (int) (unsafeNativeHandle ^ (unsafeNativeHandle >>> 32));
```

Comment:
This works too - just two heads-up:
- A new object is created each time .hashCode() is called.
- .equals() can throw exceptions when the underlying graph is deleted, but .hashCode() will never throw exceptions - pragmatically don't see any issues here other than exactly when adding/fetching items from a cache/collection with stale objects trigger exceptions.